### ISSUE TITLE
fixedlibname as default

### ISSUE THREAD

**Issue by abelsiqueira** [optimizers/CUTEst.jl#27]
The `sifdecoder` is not using the fixedlibname, so if using the default libname, the call fails.

**Comment by dpo** [comment]
Not sure I understand. Could you give an example?

**Comment by abelsiqueira** [comment]
The sifdecoder creates a libPROBLEM.so, but the interfaces use fixedlibname as default. In other words, the user needs to carry another variable (libname) along.
````julia
libname = sifdecoder("HS32")
# fortran_open call
usetup(input, out, io, n) # Won't work because assumes libname = fixedlibname
usetup(input, out, io, n, libname) # Works
````
So we either (1) make the sifdecoder use fixedlibname (optionally) or (2) forgo the default libname.

**Comment by dpo** [comment]
I see. I'm really in favor of libPROBLEM because it allows you to keep the precompiled problems around. It's also the default behavior in CUTEst. I guess the library name shouldn't be optional in the core interface.

**Comment by abelsiqueira** [comment]
Agreed. I will update PR #26 accordingly.

**Comment by abelsiqueira** [comment]
Implemented in 3f254b2.
